lib/analyzer/static-analyzer.ts

```diff
@@ -26,7 +26,11 @@ export async function analyze(
   targetImage: string,
   options: StaticAnalysisOptions,
 ) {
-  if (!options.imagePath || options.imageType === undefined) {
+  if (
+    !options.imagePath ||
+    options.imageType === undefined ||
+    options.tmpDirPath === undefined
```

Comment:
I think this is a bit harsh (making tmpDir a mandatory parameter) could we not make tmpdir() the default value and make tmpDirPath really optional?

Reply:
I think this is a good separation of concerns for the docker plugin - why burden it with having to decide and manage a temporary dir?
it's a logical component tasked with scanning an image, let its user (in this case, the Kubernetes-Monitor) provide it with storage.
this way it also won't create files in places unknown to its user.
what do you think?

Reply:
I'd prefer the caller to be explicit about their choice of directory, there's less variations happening internally in the implementation... I'm not strongly objecting having a default but since we're currently the only users of static analysis I want us to be very explicit about this choice since forgetting to set a path can really break our scanning ability

Reply:
first, this is a "breaking" change, I think it is better to have the default tmp dir in case it is not passed.
additionally it don't think it is reasonable to enforce the users to pass it, because most of them do not care about it (actually, we are the only ones who care)

Reply:
Pushing decisions to the calling code is a common but not always good pattern. If it does have severe consequences for you then yes we could make it explicit. But please do describe then what is actually expected and what do we do with it (i.e. files are left behind in the folder, folder must be empty, etc.)

Reply:
@mladkau fair enough, makes sense :) I also think we should add the description you suggested regardless of whether or not it's optional.

@shaimendel why not keep it mandatory and let the k8s-monitor upgrade to the latest docker-plugin only when it's ready to pass the new arguments? it's not a living service, you don't have to upgrade your dependency if you don't want to.

> additionally it don't think it is reasonable to enforce the users to pass it, because most of them do not care about it
aren't we the only users for static scanning?

Reply:
@Shesekino In the moment you are the only users but @snyk/magma will very soon follow. Not having static scanning is a big gap for us.

Reply:
Thanks for your input! I've made it optional, I think we can ensure we always pass the path from our side